### ISSUE TITLE
Calendar: Unable to tab in and type a date into the Calendar component

### ISSUE THREAD

**Issue by kcoltman-dev** [primefaces/primereact#8170]
### Describe the bug

Users are unable to tab through a form into a Calendar component and just type in the date. They need to click into the component before entering a date.

### Reproducer

https://codesandbox.io/p/sandbox/mystifying-wiles-qmrxr9?file=%2Fsrc%2FApp.js%3A11%2C24

### System Information

```Shell
react ^19.0.0
react-dom ^19.0.0
react-scripts ^5.0.0
primereact 10.9.6
```

### Steps to reproduce the behavior

1. Click into the Name field and enter text
2. Hit Tab
3. Attempt to enter a date (ex 01/01/2001)

### Expected behavior

Should allow date entry by keyboard without clicking into the component with a mouse

**Comment by kcoltman-dev** [comment]
https://codesandbox.io/p/sandbox/mystifying-wiles-qmrxr9?file=%2Fsrc%2FApp.js%3A11%2C24
